# `newline-per-chained-call`

Enforces a newline before each chained method call after reaching a configurable maximum depth.

## Rule details

This rule differs from the [`newline-per-chained-call`](https://eslint.org/docs/rules/newline-per-chained-call)
rule by enforcing a newline before each chained method call rather than only after the call that exceeded the maximum
depth. Another difference is that this rule only applies to chained method calls, not to chained property accessors.

## How to use

```json
{
  "@croct/newline-per-chained-call": [
    "error",
    {
      "ignoreChainWithDepth": 2
    }
  ]
}
```

## Examples

These are examples of how the rule might apply.

### ‚ùå Incorrect

```jsx
this.foo.block.bar.qux();
```

```jsx
foo().bar().baz().qux();
```

```jsx
foo.bar.baz.qux();
```

### ‚úÖ Correct

```jsx
this.foo
    .block
    .bar
    .qux();
```

```jsx
foo()
    .bar()
    .baz()
    .qux();
```

```jsx
await expect(callback).resolves.toHaveBeenCalledWith();
```

```jsx
foo.bar.baz.qux;
```

## Options

These are the available options:

### `ignoreChainDeeperThan`

Specifies the maximum depth of chained allowed, default is 2.

#### ‚ùå Incorrect

```jsx
/*eslint newline-per-chained-call: ["error", {"ignoreChainWithDepth": 1}]*/
foo().bar().baz();
```

#### ‚úÖ Correct

```jsx
/*eslint newline-per-chained-call: ["error", {"ignoreChainWithDepth": 1}]*/
foo()
    .bar()
    .baz();
```

## Attributes

- [ ] ‚úÖ Recommended
- [x] üîß Fixable
- [ ] üí≠ Requires type information
